result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -30,6 +30,23 @@ sealed public class Result<out V : Any, out E : Exception> private constructor(v
 
     }
 
+    /**
+     * provide a fallback value in case of failure
+     *
+     * It is the equivalent of the elvis operator ?: for nullable types
+     *
+     * Example:
+     *
+     * val vecsize= Result.of(vec.size() or 0)
+     *
+     */
+    public inline infix fun <reified V : Any> or(fallbackValue:V):Result<V,E> {
+        return when(this) {
+            is Success -> Success(this.value as V)
+            is Failure -> Success(fallbackValue)
+        }
+    }
```

Comment:
I have thought of the argument for 'or' as a default value, to be used if Result is of type failure, much like the elvis operator for nullable types. Does not make much sense to use a function that returns Result. But you can of course add such a function if you want. Do what you think is right in your eyes. I currently has decided to start using the library and understand what is missing from my actual usage of it.